mountaineer/paths.py

```diff
@@ -171,7 +187,7 @@ def is_root_link(self):
         return self == self.root_link
 
     @property
-    def parent(self):
+    def parent(self):  # type: ignore
```

Comment:
Even though the parent property of pathlib.Path is `-> Self`, which is the same as the other functions that we override here, pyright doesn't register that we are able to override the Self with our ManagedViewPath.